src/MiniCover/Instrumentation/Instrumenter.cs

```diff
@@ -154,13 +157,17 @@ private void InstrumentAssembly(string assemblyFile)
 
                             var instructionId = ++id;
 
-                            result.AddInstruction(sourceRelativePath, new InstrumentedInstruction
+                            result.Assemblies[assemblyDefinition.Name.Name].AddInstruction(sourceRelativePath, new InstrumentedInstruction
```

Comment:
I would prefer that the method AddInstrumentedAssembly returns the InstrumentedAssembly object.

Then we don't need to access the assembly by dictionary key here.